File: 3rd_party/adios/testing/install/C/main_mpi.c
/*
 * Distributed under the OSI-approved Apache License, Version 2.0.  See
 * accompanying file Copyright.txt for details.
 */

#include <adios2_c.h>

#if !ADIOS2_USE_MPI
#error "ADIOS2_USE_MPI is not true for source using ADIOS2 MPI bindings"
#endif

#include <mpi.h>

#include <stdio.h>

int main(int argc, char **argv)
{
    int provided;

    // MPI_THREAD_MULTIPLE is only required if you enable the SST MPI_DP
    MPI_Init_thread(&argc, &argv, MPI_THREAD_MULTIPLE, &provided);

    adios2_adios *adios = adios2_init_mpi(MPI_COMM_WORLD);
    if (!adios)
    {
        fprintf(stderr, "adios2_init() failed\n");
        return 1;
    }
    adios2_finalize(adios);

    MPI_Finalize();

    return 0;
}

File: 3rd_party/adios/testing/install/Fortran/main_mpi.f90
program adios_fortran_test
  use mpi
  use adios2
  integer :: ierr, irank, isize
  type(adios2_adios) :: adios

  call MPI_INIT(ierr)
  call MPI_COMM_RANK(MPI_COMM_WORLD, irank, ierr)
  call MPI_COMM_SIZE(MPI_COMM_WORLD, isize, ierr)

  call adios2_init(adios, MPI_COMM_WORLD, ierr)
  call adios2_finalize(adios, ierr)

  call MPI_Finalize(ierr)
end program adios_fortran_test
